File: 351002/Khmylko/rest-api/publisher/src/Controller/MarkerController.php
<?php
namespace App\Controller;

use App\Service\MarkerService;

class MarkerController {
    private MarkerService $service;

    public function __construct(MarkerService $service) {
        $this->service = $service;
    }

    public function getAll(): array {
        return $this->service->getAll();
    }

    public function getById(int $id): array {
        return $this->service->getById($id);
    }

    public function create(array $data): array {
        return $this->service->create($data);
    }

    public function update(int $id, array $data): array {
        return $this->service->update($id, $data);
    }

    public function delete(int $id): void {
        $this->service->delete($id);
    }
}

File: 351004/Student/Lab5/publisher/src/main/java/com/example/task310/controller/MarkerController.java
package com.example.task310.controller;

import com.example.task310.dto.*;
import com.example.task310.service.MarkerService;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.data.domain.Pageable;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.*;
import java.util.List;

@RestController
@RequestMapping("/api/v1.0/markers")
@RequiredArgsConstructor
public class MarkerController {
    private final MarkerService service;

    @GetMapping
    public List<MarkerResponseTo> getAll(Pageable pageable) {
        return service.getAll(pageable);
    }

    @GetMapping("/{id}")
    public MarkerResponseTo getById(@PathVariable Long id) {
        return service.getById(id);
    }

    @PostMapping
    @ResponseStatus(HttpStatus.CREATED)
    public MarkerResponseTo create(@Valid @RequestBody MarkerRequestTo dto) {
        return service.create(dto);
    }

    @PutMapping("/{id}")
    public MarkerResponseTo update(@PathVariable Long id, @Valid @RequestBody MarkerRequestTo dto) {
        dto.setId(id);
        return service.update(dto);
    }

    @DeleteMapping("/{id}")
    @ResponseStatus(HttpStatus.NO_CONTENT)
    public void delete(@PathVariable Long id) {
        service.delete(id);
    }
}